### PR TITLE
Show proper thread from notification

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/environment/Environment.m [hunk 1]
```diff
@@ -215,6 +215,24 @@ + (void)presentConversationForRecipientId:(NSString *)recipientId
...
 }
 
++ (void)presentConversationForThreadId:(NSString *)threadId
+{
+    OWSAssert(threadId.length > 0);
+
+    __block TSThread *thread;
+    [[TSStorageManager sharedManager].dbReadWriteConnection
+        readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
+            thread = [TSThread fetchObjectWithUniqueID:threadId];
```
Comment (charlesmchen) [inline Signal/src/environment/Environment.m]: What?  Shouldn't this be `[fetchObjectWithUniqueID: transaction:]` or ... why not just use `[fetchObjectWithUniqueID:]` without the wrapping transaction?

File: Signal/src/network/PushManager.m [hunk 2]
```diff
@@ -231,12 +234,21 @@ - (void)application:(UIApplication *)application
...
+        }
         completionHandler();
     } else {
         OWSFail(@"%@ Unhandled action with identifier: %@", self.tag, identifier);
         NSString *threadId = notification.userInfo[Signal_Thread_UserInfo_Key];
-        [Environment presentConversationForRecipientId:threadId];
+        if (threadId) {
+            [Environment presentConversationForThreadId:threadId];
+        } else {
+            OWSFail(@"%@ threadId was unexpectedly nil in action with identifier: %@", self.tag, identifier);
```
Comment (charlesmchen) [inline Signal/src/network/PushManager.m]: Why not DRY up this nil test into `presentConversationForThreadId:`?

Comment (michaelkirk) [reply Signal/src/network/PushManager.m]: A failure here likely indicates malformed notification params, which, being a K-V bag is easy to mess up, so I wanted to log the action as debug context.

File: Signal/src/environment/Environment.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/environment/Environment.m]: thank you!

File: Signal/src/network/PushManager.m (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/network/PushManager.m]: Makes sense.

**Pull request merged**